src/Logary/Metric/Gauge.fs

```diff
@@ -0,0 +1,79 @@
+namespace Logary.Metric
+
+
+type GaugeConf =
+  {
+   basicInfo: BasicConf
+   histogramConf: HistogramConf option
+  }
+
+  interface MetricBuilder<IGauge> with
```

Comment:
IGauge is something like 
https://godoc.org/github.com/prometheus/client_golang/prometheus#Gauge 

`MetricBuilder<IGauge>` here means this conf can produce a `IGauge` metric, this can give one conf to produce different `IMetric` , if their conf are same. (e.g. `counter`, `gauge`) , which will be use depend on the type inference